File: app/src/main/java/linusfessler/alarmtiles/sleeptimer/SleepTimerConfigFragment.java
package linusfessler.alarmtiles.sleeptimer;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.Fragment;

import javax.inject.Inject;

import io.reactivex.disposables.CompositeDisposable;
import linusfessler.alarmtiles.R;
import linusfessler.alarmtiles.core.App;
import linusfessler.alarmtiles.databinding.FragmentSleepTimerConfigBinding;

public class SleepTimerConfigFragment extends Fragment {

    @Inject
    SleepTimerViewModel viewModel;

    private FragmentSleepTimerConfigBinding binding;

    private final CompositeDisposable disposable = new CompositeDisposable();

    @Override
    public void onCreate(@Nullable final Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ((App) requireActivity().getApplicationContext())
                .getAppComponent()
                .inject(this);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull final LayoutInflater inflater, @Nullable final ViewGroup container, @Nullable final Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);

        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_sleep_timer_config, container, false);

        disposable.add(viewModel.getSleepTimer()
                // Initialize view, afterwards it matches the user input already
                .firstElement()
                .subscribe(sleepTimer -> {
                    binding.decreaseVolume.setChecked(sleepTimer.isDecreasingVolume());
                    binding.decreaseVolume.setOnCheckedChangeListener((buttonView, isChecked) ->
                            viewModel.dispatch(new SleepTimerEvent.SetDecreasingVolume(isChecked)));
                }));

        return binding.getRoot();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        disposable.clear();
        binding.decreaseVolume.setOnCheckedChangeListener(null);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        disposable.dispose();
    }
}

File: app/src/main/java/linusfessler/alarmtiles/sleeptimer/SleepTimerConfigFragment.kt
package linusfessler.alarmtiles.sleeptimer

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import io.reactivex.disposables.CompositeDisposable
import linusfessler.alarmtiles.core.App
import linusfessler.alarmtiles.databinding.FragmentSleepTimerConfigBinding
import javax.inject.Inject

class SleepTimerConfigFragment : Fragment() {
    @Inject
    lateinit var viewModel: SleepTimerViewModel

    private lateinit var binding: FragmentSleepTimerConfigBinding

    private val disposable = CompositeDisposable()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        (requireActivity().applicationContext as App)
                .sleepTimerComponent
                .inject(this)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        binding = FragmentSleepTimerConfigBinding.inflate(inflater, container, false)

        // Initialize view, afterwards it will always match the user input
        disposable.add(viewModel.sleepTimer
                .firstElement()
                .subscribe {
                    binding.decreaseVolume.isChecked = it.isDecreasingVolume
                    binding.decreaseVolume.setOnCheckedChangeListener { _, isChecked ->
                        viewModel.dispatch(SleepTimerEvent.SetDecreasingVolume(isChecked))
                    }
                })

        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        disposable.clear()
        binding.decreaseVolume.setOnCheckedChangeListener(null)
    }

    override fun onDestroy() {
        super.onDestroy()
        disposable.dispose()
    }
}